pkg/serverless/trace/inferredspan/span_enrichment.go

```diff
@@ -217,10 +220,120 @@ func (inferredSpan *InferredSpan) EnrichInferredSpanWithS3Event(eventPayload eve
 	}
 }
 
+type DatadogPayload struct {
+	Type        string `json:"Type"`
+	Value       string `json:"Value"`
+	DataType    string `json:"dataType"`
+	BinaryValue string `json:"binaryValue"`
+	StringValue string `json:"stringValue"`
+}
+
+type TraceHeader struct {
+	TraceID          string `json:"x-datadog-trace-id"`
+	ParentID         string `json:"x-datadog-parent-id"`
+	SamplingPriority string `json:"x-datadog-sampling-priority"`
+	Sampled          string `json:"x-datadog-sampled"`
+}
+
+type customMessageAttributeStruct struct {
+	Type  string `json:"Type"`
+	Value string `json:"Value"`
+}
+type BodyStruct struct {
+	Message           string                                  `json:"Message"`
+	MessageAttributes map[string]customMessageAttributeStruct `json:"MessageAttributes"`
+}
+type CustomDatadogPayload struct {
+	Type        string `json:"Type"`
+	Value       string `json:"Value"`
+	DataType    string `json:"dataType"`
+	BinaryValue string `json:"binaryValue"`
+	StringValue string `json:"stringValue"`
+}
+
+func extractContextFromSNSSQSEvent(firstRecord events.SQSMessage) (string, string, string, error) {
+	var bodyStruct BodyStruct
+	if err := json.Unmarshal([]byte(firstRecord.Body), &bodyStruct); err != nil {
+		return "", "", "", err
+	}
+	var ddCustomPayloadValue, ok = bodyStruct.MessageAttributes["_datadog"]
+	if !ok {
+		return "", "", "", errors.New("no Datadog trace context found")
```

Comment:
Hmmm will this error message be displayed on every single message where there is no datadog context in an SQS message? If so, we should suppress it.